src/txn/txn_recover.c

```diff
@@ -503,6 +504,36 @@ __recovery_set_ckpt_base_write_gen(WT_RECOVERY *r)
     return (ret);
 }
 
+/*
+ * __recovery_txn_setup_initial_state --
+ *     Setup the transaction initial state required by rollback to stable.
+ */
+static int
+__recovery_txn_setup_initial_state(WT_SESSION_IMPL *session, WT_RECOVERY *r)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
+    WT_RET(__recovery_set_checkpoint_snapshot(session));
+    WT_RET(__recovery_set_checkpoint_timestamp(r));
+    WT_RET(__recovery_set_oldest_timestamp(r));
+
+    /* Update the global time window state to have consistent view from global visibility rules */
+    __wt_txn_update_pinned_timestamp(session, true);
+
+    WT_ASSERT(session,
+      conn->txn_global.has_stable_timestamp == false &&
+        conn->txn_global.stable_timestamp == WT_TS_NONE);
+
+    /* Set the stable timestamp from recovery timestamp. */
+    conn->txn_global.stable_timestamp = conn->txn_global.recovery_timestamp;
+    if (conn->txn_global.recovery_timestamp != WT_TS_NONE)
```

Comment:
could we change this condition as "if (conn->txn_global.stable_timestamp != WT_TS_NONE)" basically check the value of stable timestamp itself, instead of checking the recovery timestamp

Reply:
Yes, fixed. Thanks.